python_modules/dagster-graphql/dagster_graphql/schema/runs.py

```diff
@@ -211,6 +211,28 @@ def parse_run_config_input(
     return run_config
 
 
+class GrapheneMegaRuns(graphene.ObjectType):
+    results = non_null_list("dagster_graphql.schema.pipelines.pipeline.GrapheneMegaRun")
+
+    class Meta:
+        name = "MegaRuns"
+
+    def __init__(self, cursor, limit):
+        super().__init__()
+
+        self._cursor = cursor
+        self._limit = limit
+
+    def resolve_results(self, graphene_info: ResolveInfo):
+        return get_mega_runs(graphene_info, self._cursor, self._limit)
+
```

Comment:
having an intermediate object here we should model this more as a "connection" than just a plural of results. Basically this just means returning cursor and maybe "has more" here. Also might inform a slightly different name. We are not consistent about this but if you grep for "connection" you can see some examples. 

https://graphql.org/learn/pagination/#complete-connection-model